app/controllers/two_factor_authentication/backup_code_verification_controller.rb

```diff
@@ -0,0 +1,76 @@
+module TwoFactorAuthentication
+  class BackupCodeVerificationController < ApplicationController
+    include TwoFactorAuthenticatable
+
+    prepend_before_action :authenticate_user
+    prepend_before_action :handle_if_all_codes_used
+
+    # rubocop:disable Layout/FirstParameterIndentation
+    def show
+      analytics.track_event(
+          Analytics::MULTI_FACTOR_AUTH_ENTER_BACKUP_CODE_VISIT, context: context
+        )
+      @presenter = TwoFactorAuthCode::BackupCodePresenter.new(
+          view: view_context,
+          data: { current_user: current_user }
+        )
+      @backup_code_form = BackupCodeVerificationForm.new(current_user)
+    end
+
+    def create
+      @backup_code_form = BackupCodeVerificationForm.new(current_user)
+      result = @backup_code_form.submit(backup_code_params)
+      analytics.track_event(Analytics::MULTI_FACTOR_AUTH, result.to_h)
+
+      handle_result(result)
+    end
+
```

Comment:
How about moving 'private' here?

Reply:
Done

Reply:
private should be before handle_if_all_codes_used